### PR TITLE
Change execution courses table in teacher portal to be ordered by period and name. ACDM-1378 #resolve

### PR DESCRIPTION
**Pull request by francisconeves97**

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java [hunk 1]
```diff
@@ -128,6 +128,21 @@ public int compare(ExecutionCourse o1, ExecutionCourse o2) {
...
+                    final int cep = o2.getExecutionPeriod().compareTo(o1.getExecutionPeriod());
+                    if (cep != 0) {
+                        return cep;
+                    }
+                    final int c = Collator.getInstance().compare(o1.getNome(), o2.getNome());
+                    return c == 0 ? DomainObjectUtil.COMPARATOR_BY_ID.compare(o1, o2) : c;
+                }
+
+            };
+
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java]: This code can be replaced to use `EXECUTION_COURSE_EXECUTION_PERIOD_COMPARATOR` and `EXECUTION_COURSE_NAME_COMPARATOR`. `Comparator.reversed()` should help.
Maybe you can also change the implementation of `EXECUTION_COURSE_COMPARATOR_BY_EXECUTION_PERIOD_AND_NAME` accordingly.

Comment (francisconeves97) [reply src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java]: Changed both the implementations accordingly.

File: src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java [hunk 2]
```diff
@@ -118,11 +118,26 @@ public int compare(ExecutionCourse o1, ExecutionCourse o2) {
...
                         return cep;
                     }
-                    final int c = Collator.getInstance().compare(o1.getNome(), o2.getNome());
+                    final int c = EXECUTION_COURSE_NAME_COMPARATOR.compare(o1, o2);
+                    return c == 0 ? DomainObjectUtil.COMPARATOR_BY_ID.compare(o1, o2) : c;
+                }
+
+            };
+
+    public static final Comparator<ExecutionCourse> EXECUTION_COURSE_COMPARATOR_BY_EXECUTION_PERIOD_REVERSED_AND_NAME =
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java]: check if ``Comparator.reversed()`` can be used instead of this comparator.

Comment (francisconeves97) [reply src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java]: `Comparator.reversed()` can be used but the ExecutionCourse order will also be reversed alphabetically.

**Pull request merged**